Python/sysmodule.c

```diff
@@ -508,6 +539,10 @@ sys_audit(PyObject *self, PyObject *const *args, Py_ssize_t argc)
     }
 
     if (!should_audit(tstate->interp)) {
+        // Validate first argument using assertions, only impact peformance
+        // of Python built in debug mode.
+        assert(args[0] != NULL);
+        assert(PyUnicode_Check(args[0]));
```

Comment:
Better to assert in all calls in debug builds, so that results don't vary based on whether a hook is enabled or not.

Reply:
Ok, I moved the assertions to also run tem if should_audit() is true.